### PR TITLE
feat: events on a denylist don't update person properties

### PR DESCRIPTION
**Pull request by tiina303**
https://posthog.slack.com/archives/C0374DA782U/p1719318556646959?thread_ts=1719222351.040789&cid=C0374DA782U

Exception events and $$heatmap events will start going to a different pipeline, which might be out of sync with analytics pipeline, we don't want it to override properties to older values from here if it that pipeline were to lag. Exception events don't need to update person properties, analytics events do.

We'll still use `$set` to override the values for PoE for the exception event.

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
👍

**Review by bretthoerner — APPROVED** [review]
Looks good other than test (lint) being mad

**Comment by tiina303** [comment]
> [warn] src/worker/ingestion/person-state.ts
[warn] Code style issues found in the above file. Forgot to run Prettier?

What happens when you write code in github UI 😅

**Comment by pauldambra** [comment]
that method mutates its parameter, which scares me, so i moved to an early return to avoid risking leaking the updates

the tests for person state wouldn't run locally for me, so i added some but i don't know if they pass

File: plugin-server/src/worker/ingestion/person-state.ts [hunk 1]
```diff
@@ -348,6 +351,13 @@ export class PersonState {
      * @returns true if the properties were changed, false if they were not
      */
     private applyEventPropertyUpdates(personProperties: Properties): boolean {
+        // this relies on making changes to the object instance, so...
+        // if we should not update the person,
+        // we return early before changing any values
+        if (NO_PERSON_UPDATE_EVENTS.has(this.event.event)) {
+            return false
+        }
```
Comment (tiina303) [inline plugin-server/src/worker/ingestion/person-state.ts:359]: The problem with moving this here is that the browser etc on the event could now be wrong, but folks shouldn't be using that anyway, so let's ship it and update when we re-work the pipeline

**Pull request merged**